### ISSUE TITLE
Docker-compose log streaming crash when containers stops or restarts

### ISSUE THREAD

**Issue by fgiovatreedom** [docker/compose#7932]
## Description of the issue
Docker-compose log streaming crash when containers stops or restarts.

## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020

```

**Output of `docker version`**
```
Client: Docker Engine - Community
 Cloud integration: 1.0.2
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker-compose config`**
```
networks:
  default:
    external: true
    name: development
services:
  node:
    command:
    - npm
    - run
    - dev-new
    environment:
      NODE_ENV: development
    image: node:dubnium
    networks:
      default:
        aliases:
        - node.localhost
    ports:
    - published: 80
      target: 8080
    volumes:
    - ./node:/home/app:rw
    working_dir: /home/app
```

## Steps to reproduce the issue

1. Run ```docker-compose up```
2. Then change shell and run 
```
docker-compose stop; docker-compose rm; docker-compose up -d 
```
or 
```
docker-compose restart
```

### Observed result
On shell with logs attached:
```
node_1 exited with code 0
Exception in thread Thread-25:
Traceback (most recent call last):
  File "site-packages/urllib3/response.py", line 696, in _update_chunk_length
ValueError: invalid literal for int() with base 16: b''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages/urllib3/response.py", line 436, in _error_catcher
  File "site-packages/urllib3/response.py", line 763, in read_chunked
  File "site-packages/urllib3/response.py", line 700, in _update_chunk_length
http.client.IncompleteRead: IncompleteRead(0 bytes read)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages/requests/models.py", line 751, in generate
  File "site-packages/urllib3/response.py", line 571, in stream
  File "site-packages/urllib3/response.py", line 792, in read_chunked
  File "contextlib.py", line 130, in __exit__
  File "site-packages/urllib3/response.py", line 454, in _error_catcher
urllib3.exceptions.ProtocolError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "threading.py", line 926, in _bootstrap_inner
  File "threading.py", line 870, in run
  File "compose/cli/log_printer.py", line 214, in watch_events
  File "compose/project.py", line 575, in yield_loop
  File "compose/project.py", line 543, in build_container_event
  File "compose/container.py", line 44, in from_id
  File "site-packages/docker/utils/decorators.py", line 19, in wrapped
  File "site-packages/docker/api/container.py", line 771, in inspect_container
  File "site-packages/docker/utils/decorators.py", line 46, in inner
  File "site-packages/docker/api/client.py", line 228, in _get
  File "site-packages/requests/sessions.py", line 543, in get
  File "site-packages/requests/sessions.py", line 530, in request
  File "site-packages/requests/sessions.py", line 685, in send
  File "site-packages/requests/models.py", line 829, in content
  File "site-packages/requests/models.py", line 754, in generate
requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))
```
And after that the log stream remain detached

### Expected result
```
node_1 exited with code 0
node_1            | 
node_1            | > node@0.0.1 dev-new /home/app
node_1            | > npm install --no-package-lock; nodemon --delay 1 ./bin/www --exec babel-node --presets env
node_1            | 
....
```
The log stream must be remain attached

### Stacktrace / full error message

```
Exception in thread Thread-25:
Traceback (most recent call last):
  File "site-packages/urllib3/response.py", line 696, in _update_chunk_length
ValueError: invalid literal for int() with base 16: b''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages/urllib3/response.py", line 436, in _error_catcher
  File "site-packages/urllib3/response.py", line 763, in read_chunked
  File "site-packages/urllib3/response.py", line 700, in _update_chunk_length
http.client.IncompleteRead: IncompleteRead(0 bytes read)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages/requests/models.py", line 751, in generate
  File "site-packages/urllib3/response.py", line 571, in stream
  File "site-packages/urllib3/response.py", line 792, in read_chunked
  File "contextlib.py", line 130, in __exit__
  File "site-packages/urllib3/response.py", line 454, in _error_catcher
urllib3.exceptions.ProtocolError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "threading.py", line 926, in _bootstrap_inner
  File "threading.py", line 870, in run
  File "compose/cli/log_printer.py", line 214, in watch_events
  File "compose/project.py", line 575, in yield_loop
  File "compose/project.py", line 543, in build_container_event
  File "compose/container.py", line 44, in from_id
  File "site-packages/docker/utils/decorators.py", line 19, in wrapped
  File "site-packages/docker/api/container.py", line 771, in inspect_container
  File "site-packages/docker/utils/decorators.py", line 46, in inner
  File "site-packages/docker/api/client.py", line 228, in _get
  File "site-packages/requests/sessions.py", line 543, in get
  File "site-packages/requests/sessions.py", line 530, in request
  File "site-packages/requests/sessions.py", line 685, in send
  File "site-packages/requests/models.py", line 829, in content
  File "site-packages/requests/models.py", line 754, in generate
requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))
```

## Additional information
I'm running docker on macOS Catalina (10.15.7)

**Comment by grit96** [comment]
I have the same issue on macOS Big Sur - the logs crash and don't reattach when containers are restarted.

**Comment by blixt** [comment]
I'm reliably seeing this on macOS 11.0.1 (20B29) with:

```
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

---

```
Client: Docker Engine - Community
 Cloud integration: 1.0.2
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Comment by thaJeztah** [comment]
Wondering if this is related to https://github.com/docker/docker-py/issues/2696 /  https://github.com/docker/for-mac/issues/5059#issuecomment-726036933

@aiordache @simonferquel

**Comment by BrianGilbert** [comment]
Also getting this with Big Sur 11.0.1 (20B29) 

Issue is present with stable or edge version of Docker.

**Comment by kelseym** [comment]
I am seeing the same error on Catalina 10.15.7
Docker v19.03.13
docker-compose version 1.27.4, build 40524192

**Comment by thaJeztah** [comment]
Does the problem go away if you make compose use the "raw" socket? You can set the `DOCKER_HOST` environment variable to switch;

```bash
export DOCKER_HOST=unix://"$HOME"/Library/Containers/com.docker.docker/Data/docker.raw.sock
```

And to unset that variable;

```bash
unset DOCKER_HOST
```

**Comment by simonferquel** [comment]
Note, @thaJeztah is only to narrow the problem down a little bit, you should not mount the raw socket in a container, as it won't be able to handle bind-mounts of your mac files.

**Comment by thaJeztah** [comment]
Ah, yes, read my "big fat warning" here; https://github.com/docker/docker-py/issues/2696#issuecomment-729636008

**Comment by Arashtbr** [comment]
I have the same error on Catalina Version 10.15.4
Docker v19.03.13 &
```docker-compose version 1.27.4, build 40524192```

**Comment by BrianGilbert** [comment]
This seems to be resolved as of edge 2.5.2 which came out today
